### PR TITLE
Add documentation for Delta Lake configuration properties

### PR DESCRIPTION
**Pull request by findinpath**
Add the configuration properties to the general configuration properties section to have all of them centralised.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -139,6 +139,10 @@ values. Typical usage does not require you to configure them.
       - Enables :ref:`Table statistics <delta-lake-table-statistics>` for
         performance improvements.
       - ``true``
+    * - ``delta.extended-statistics.enabled``
+      - Enable collection through :doc:`/sql/analyze` and
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:143]:
```suggestion
      - Enable statistics collection with :doc:`/sql/analyze` and
```

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -159,6 +163,9 @@ values. Typical usage does not require you to configure them.
     * - ``delta.register-table-procedure.enabled``
       - Enable to allow users to call the ``register_table`` procedure.
       - ``false``
+    * - ``delta.vacuum.min-retention``
+      - Minimal retention period for the :ref:`VACUUM<delta-lake-vacuum>` procedure.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:167]: what is this actually ... the minimum duration (link to prop type) or age of files of ?? .. younger files are removed by vacuum?

**Review by mosabua — CHANGES_REQUESTED** [review]
This will also collide with the session properties PR

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 2 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/delta-lake.rst]: Added more information to avoid misleading the users about this configuration property.

**Comment by findinpath** [comment]
> This will also collide with the session properties PR

I see no problem here - will adjust eventual conflicts as soon as one of the PRs lands.

**Review by mosabua — APPROVED** [review]

**Comment by mosabua** [comment]
Verified doc build locally. CI failure is unrelated. Merging.

**Pull request merged**